terraform/cp4ba/main.tf

```diff
@@ -0,0 +1,179 @@
+# Provider block
+provider "ibm" {
+  region           = var.region
+  version          = "~> 1.12.0"
+  ibmcloud_api_key = var.ibmcloud_api_key
+}
+
+
+// user output and pwd outputs
+
+# Getting the OpenShift cluster configuration
+data "ibm_resource_group" "group" {
+  name = var.resource_group
+}
+
+resource "null_resource" "mkdir_kubeconfig_dir" {
+  triggers = { always_run = timestamp() }
+  provisioner "local-exec" {
+    command = "mkdir -p ${var.config_dir}"
+  }
+}
+
+data "ibm_container_cluster_config" "cluster_config" {
+  depends_on = [null_resource.mkdir_kubeconfig_dir]
+  cluster_name_id   = var.cluster_name_id
+  resource_group_id = data.ibm_resource_group.group.id
+  download          = true
+  config_dir        = "./kube/config"
+  admin             = false
+  network           = false
+}
+
+
+module "cluster" {
+  source = "git::https://github.com/ibm-hcbt/terraform-ibm-cloud-pak/tree/main/modules/roks"
+//  source = "../../roks"
+  // source = "git::https://github.com/ibm-hcbt/terraform-ibm-cloud-pak.git//modules/roks"
+  enable = local.enable_cluster
+//  on_vpc = var.on_vpc
+
+  project_name             = var.project_name
+  owner                    = var.entitled_registry_user_email
+  environment              = var.environment
+
+  resource_group       = var.resource_group
+  roks_version         = var.platform_version
+  flavors              = var.flavors
+  workers_count        = var.workers_count
+  datacenter           = var.data_center
+  force_delete_storage = true
```

Comment:
I would not hard code this value.  Make it a variable that the user has to set.

Reply:
Ok, I will.